### PR TITLE
Refactoring FT GHA Workflows

### PR DESCRIPTION
**Pull request by kairoaraujo**
It adds some refactoring of the functional test GitHub Actions from the umbrella. It requires the workflows to deploy the RSTUF using docker-compose and informing the CLI version.

It will allow components to run FT after a PR is approved.

The components must use docker compose for the containers using resources as they want. Multiple docker-compose.yml might be interesting to cover different RSTUF deployments scenarios. It requires following the normal expected steps of setting up the RSTUF.

The components must check the main repository-service-tuf (umbrella) repository as `rstuf-umbrella` and can use the `tests/functions/scripts`.

The `run-ft` starts the Functional Tests using different Bootstrap/Setup.

Extend the timeout values for performance.

The tests running inside the container have low performance if compared running from the runners

To summarize:

- Components deploy the RSTUF using Docker Compose (multiple can be done in different jobs), allowing different deployments for the component.

- Components run the `rstuf-umbrella/tests/function/run-tf*` using different Bootstrap/Ceremony methods that are global to all components.

### CODE REVIEW

File: .github/workflows/functional.yml [hunk 1]
```diff
@@ -2,62 +2,94 @@ name: Functional Tests
...
-      RSTUF_ONLINE_KEY: ${{ secrets.RSTUF_ONLINE_KEY }}
+  functional-das:
+    name: "DAS"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout RSTUF Umbrella
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+
+      - name: RSTUF Deployment
```
Comment (MVrachev) [inline .github/workflows/functional.yml:61]: Nit:
```suggestion
      - name: RSTUF Deployment with docker compose
```

File: .github/workflows/functional.yml [hunk 2]
```diff
@@ -2,62 +2,94 @@ name: Functional Tests
...
+        uses: isbang/compose-action@178aeba5c9dbeed89ffffbb3e6548ec08e9839cf
+        with:
+          compose-file: ${{ inputs.docker_compose }}
+        env:
+          API_VERSION: ${{ inputs.api_version }}
+          WORKER_VERSION: ${{ inputs.worker_version }}
+          RSTUF_ONLINE_KEY: ${{ secrets.RSTUF_ONLINE_KEY }}
```
Comment (MVrachev) [inline .github/workflows/functional.yml:90]: I will add a comment that if `RSTUF_ONLINE_KEY` is used in docker-compose, then it should be defined in the repository used for prs.

File: .github/workflows/functional.yml [hunk 3]
```diff
@@ -2,62 +2,94 @@ name: Functional Tests
...
```
Comment (MVrachev) [inline .github/workflows/functional.yml:90]:
```suggestion
          # If you use "RSTUF_ONLINE_KEY" make sure it's
          # defined in the repositories you use for prs.
          RSTUF_ONLINE_KEY: ${{ secrets.RSTUF_ONLINE_KEY }}
```

**Review by MVrachev — APPROVED** [review]
